include/ur_api.h

```diff
@@ -1562,6 +1562,10 @@ typedef enum ur_device_info_t {
                                                                     ///< semaphore resources
     UR_DEVICE_INFO_INTEROP_SEMAPHORE_EXPORT_SUPPORT_EXP = 0x200F,   ///< [::ur_bool_t] returns true if the device supports exporting internal
                                                                     ///< event resources
+    UR_DEVICE_INFO_COMPONENT_DEVICES = 0x2010,                      ///< [::ur_device_handle_t[]] returns the set of component devices contained
```

Comment:
Could you add these to the spec yml here https://github.com/oneapi-src/unified-runtime/blob/dcec3feb7912be6e6fb1a8ef268b2d61afe759c5/scripts/core/device.yml#L153 and run the `generate` target to make these changes? It would also be useful to have these in the DeviceGetInfo test https://github.com/oneapi-src/unified-runtime/blob/main/test/conformance/device/urDeviceGetInfo.cpp#L114

Reply:
@aarongreig Sure, I can do that, but I'm a bit confused because I don't see the immediately preceding one (`UR_DEVICE_INFO_INTEROP_SEMAPHORE_EXPORT_SUPPORT_EXP`) in that file. Also, I just tried `make generate` and it requires `clang-format-15`, is that right? Or is it 15 or newer?

Reply:
Ah right, yeah the _EXP enums are generated from separate EXP (for experimental) yml files like [this one](https://github.com/oneapi-src/unified-runtime/blob/67e4d1bcf0057eef7910851bfb5020df9e5befd6/scripts/core/exp-bindless-images.yml#L83), whereas your change is going directly into the core spec. The generate scripts will figure out the order and stuff for you.

Yeah we do require clang-format-15 just to prevent noisy diffs due to changes in the tool's opinion on things between versions

Reply:
@aarongreig this PR's goal is to support a new extension, it's not a core SYCL feature. Just pointing it out in case it matters  in terms of what file I should update. I'm going for `device.yml` unless you say otherwise. Thank you for your help!

Reply:
I just added a commit with the changes made by `make generate` but, as you can see, it changed the whole `/include/ur_print.hpp` file. Did I do it wrong?

Reply:
it looks (based on the CI) like your `generate` run formatted a lot of `ur_print.hpp` differently, maybe cmake picked up the wrong clang-format version?

Reply:
I don't think so, I didn't have `clang-format-15` and cmake complained about it. I installed it, and after that, it worked, and I could run `make generate`. I just double checked and the output from cmake includes this:

`-- Found clang-format: /iusers2/mmaronas/tools/llvm-project/install/bin/clang-format (version: 15.0)`.

Reply:
the build is failing because of the formatting changes. You need to have exactly the right version - `clang-format-15.0.7`. This is included in python prerequisites, so you should have it if you follow instructions [here](https://oneapi-src.github.io/unified-runtime/core/CONTRIB.html#build-environment).

Reply:
@pbalcer I have installed the python prerequisites as instructed:
`-- Found clang-format: /iusers2/mmaronas/sycl_workspace/llvm-bld/_deps/unified-runtime-src/.local/bin/clang-format (version: 15.0)`.
 Then, ran `make generate` again and pushed the results in https://github.com/oneapi-src/unified-runtime/pull/1192/commits/84a8b392ebdb08e10fe224390769b19eb5d20f35. It didn't change a single line in `ur_print.hpp`.

Reply:
maybe the format step is somehow silently failing to run, could you try formatting the changes that `generate` makes manually to see if what you get is any different?

Reply:
@aarongreig I think you might be right. I think the problem is that the target doesn't complete because I don't have `doxygen`, so it never runs clang-format. I'll try and get `doxygen` and see if that fixes it. Thanks both (@pbalcer) for your help!

Reply:
@aarongreig @pbalcer I could run `generate` to completion, and now the diff looks reasonable. I also updated the test you mentioned.